arm-insights/2015-07-01/swagger/insightsManagementClient_DiagnosticsSettings.json

```diff
@@ -59,7 +59,7 @@
             "in": "body",
             "required": true,
             "schema": {
-              "$ref": "#/definitions/ServiceDiagnosticSettingsCreateOrUpdateParameters"
+              "$ref": "#/definitions/ServiceDiagnosticSettingsResource"
             },
             "description": "Parameters supplied to the operation."
           },
```

Comment:
@gucalder @andyshen Has the service fixed the bug of requiring the user to provide name (Resource.name) in the body? It is a very easy fix and it will be a non breaking change. If the previous clients send the name then you can definitely ignore it. The client is supposed to give you the name of the resource in the path/url. Once you do this fix,  Resource.name can be set to `readOnly:true`